File: frontend/app/stores/interview.ts
// The interview simulator's client state — the one place a practice session's current
// question, its last coaching and its readiness live while the panel is open.
//
// It mirrors stores/chat.ts (options API: typed state, getters, async actions) and holds
// server data for the same reason: a session is a loop — ask, answer, grade, adapt — and
// each turn's result has to accumulate somewhere a component can render. Everything here
// is derived from the server; the store never decides an outcome, it records the one the
// server returned.
//
// The phase's rule is visible in what this store does NOT hold. An answer's evaluation is
// coaching only — dimensions, strengths, improvements, a suggested answer — with no
// probability and no verdict, because `InterviewAnswerEvaluation` has no such field. And
// readiness is never taken from a submit or a summary the model wrote: it is the
// platform's own signal, fetched from its own endpoint after every grade, so a completed
// session's readiness is computed, not authored. Practice, never prediction
// (docs/INTERVIEW_SIMULATOR.md).
import { defineStore } from 'pinia'
import type {
  CandidateProfile,
  InterviewAnswer,
  InterviewAnswerEvaluation,
  InterviewAnswerOutcome,
  InterviewMode,
  InterviewQuestion,
  InterviewSession,
  InterviewSessionDetail,
  InterviewSessionList,
  InterviewSessionSummary,
  InterviewSessionSummaryList,
  InterviewTurn,
  SessionReadiness,
} from '~/types/v2'
import { apiGet, apiPost, apiPostForm } from '~/utils/api-client'
import {
  interviewEvaluate,
  interviewSession as interviewSessionPath,
  V2_ENDPOINTS,
} from '~/utils/endpoints'
import { errorMessage } from '~/utils/v2-errors'

interface InterviewState {
  // The account's practice sessions, newest first — the left-hand list.
  sessions: InterviewSession[]
  // Completed sessions' readiness summaries — the trend the platform kept, not a forecast.
  history: InterviewSessionSummary[]
  // The candidate profile a new session grounds on; null until loaded (or none saved).
  profileId: string | null
  // The open session and everything derived from it while its panel is up.
  activeId: string | null
  session: InterviewSession | null
  // The question awaiting an answer, if any; null at CREATED and after each answer.
  question: InterviewQuestion | null
  // The most recently answered question — kept so its coaching, and its sequence for a
  // strict re-grade, survive the moment `question` is cleared.
  lastQuestion: InterviewQuestion | null
  lastAnswer: InterviewAnswer | null
  lastEvaluation: InterviewAnswerEvaluation | null
  // The platform's readiness signal for the open session — fetched from its own endpoint.
  readiness: SessionReadiness | null
  // The finalizing summary, once the session is completed.
  summary: InterviewSessionSummary | null
  // `loading` is a panel switch (fetching a session's detail); `busy` is an action in
  // flight (ask, answer, grade, complete, abandon) so a component can disable its controls.
  loading: boolean
  busy: boolean
  error: string | null
}

export const useInterviewStore = defineStore('interview', {
  state: (): InterviewState => ({
    sessions: [],
    history: [],
    profileId: null,
    activeId: null,
    session: null,
    question: null,
    lastQuestion: null,
    lastAnswer: null,
    lastEvaluation: null,
    readiness: null,
    summary: null,
    loading: false,
    busy: false,
    error: null,
  }),
  getters: {
    // Open for practice: the session exists and has not been completed or abandoned.
    isActive: (state): boolean => state.session?.is_active === true,
    // The loop is waiting on the candidate: a question is on the table.
    awaitingAnswer: (state): boolean => state.question !== null,
    // Finalized — the summary and its computed readiness are the last word.
    isComplete: (state): boolean => state.session?.status === 'COMPLETED',
    // A next question can be asked when the session is open and nothing is pending.
    canAsk (state): boolean {
      return state.session?.is_active === true && state.question === null
    },
  },
  actions: {
    /**
     * Resolve the account's candidate profile id — what a new session grounds on.
     *
     * A missing profile is an expected empty state, not an error to shout about: the id
     * stays null and the page steers the user to save one. Any other fault is swallowed
     * too, because a create that needs the id will surface the real refusal itself.
     */
    async loadProfile (): Promise<void> {
      try {
        const profile = await apiGet<CandidateProfile>(V2_ENDPOINTS.profile)
        this.profileId = profile.id
      }
      catch {
        this.profileId = null
      }
    },

    /** Load this account's sessions, newest first (the server orders them). */
    async loadSessions (): Promise<void> {
      try {
        const list = await apiGet<InterviewSessionList>(V2_ENDPOINTS.interviewSessions)
        this.sessions = list.sessions
      }
      catch (caught) {
        this.error = errorMessage(caught)
      }
    },

    /** Load the readiness history — completed sessions' computed signals, the trend line. */
    async loadHistory (): Promise<void> {
      try {
        const list = await apiGet<InterviewSessionSummaryList>(V2_ENDPOINTS.interviewHistory)
        this.history = list.summaries
      }
      catch (caught) {
        this.error = errorMessage(caught)
      }
    },

    // LOAD-MARKER

    /**
     * Open a new practice session over one opportunity, then select it.
     *
     * The profile id is required in the body — without it there is nothing to ground on,
     * so the store refuses before any request and points the user at their profile. The
     * server does the real grounding (profile owner-first, posting by id) and answers 404
     * if either is missing; that refusal flows back through `error`.
     */
    async create (opportunityId: string, mode: InterviewMode): Promise<string | null> {
      if (this.profileId === null) {
        this.error = 'Save a profile before starting a practice session.'
        return null
      }
      if (this.busy) return null
      this.busy = true
      this.error = null
      try {
        const session = await apiPost<InterviewSession>(V2_ENDPOINTS.interviewSessions, {
          candidate_profile_id: this.profileId,
          opportunity_id: opportunityId,
          mode,
        })
        this.sessions = [session, ...this.sessions]
        this.activeId = session.id
        this.session = session
        this.question = null
        this.lastQuestion = null
        this.lastAnswer = null
        this.lastEvaluation = null
        this.readiness = null
        this.summary = null
        return session.id
      }
      catch (caught) {
        this.error = errorMessage(caught)
        return null
      }
      finally {
        this.busy = false
      }
    },

    // SELECT-MARKER

    /**
     * Open an existing session: load its detail, derive where the loop stands, and fetch
     * the platform's live readiness.
     *
     * "Where the loop stands" is read, not remembered: the current question is the one
     * question with no answer, and the last coaching is the evaluation tied to the last
     * answer. Readiness is fetched from its own endpoint rather than taken from the
     * detail, because it is the platform's computed signal — the same rule a completed
     * session obeys.
     */
    async select (id: string): Promise<void> {
      this.activeId = id
      this.resetActive()
      this.loading = true
      try {
        const detail = await apiGet<InterviewSessionDetail>(
          interviewSessionPath(V2_ENDPOINTS.interviewSessionDetail, id))
        this.session = detail.session
        this.summary = detail.summary
        const answered = new Set(detail.answers.map(answer => answer.question_id))
        this.question = detail.questions.find(question => !answered.has(question.id)) ?? null
        const lastAnswer = detail.answers.at(-1) ?? null
        this.lastAnswer = lastAnswer
        this.lastQuestion = lastAnswer === null
          ? null
          : detail.questions.find(question => question.id === lastAnswer.question_id) ?? null
        this.lastEvaluation = lastAnswer === null
          ? null
          : detail.evaluations.find(evaluation => evaluation.answer_id === lastAnswer.id) ?? null
        this.readiness = await apiGet<SessionReadiness>(
          interviewSessionPath(V2_ENDPOINTS.interviewReadiness, id))
      }
      catch (caught) {
        this.error = errorMessage(caught)
      }
      finally {
        this.loading = false
      }
    },

    // TURN-MARKER

    /**
     * Ask the next question — discharge the plan's next topic, or adapt to the last answer.
     *
     * The server owns the choice; the store only records the turn it returns and the
     * session it moved (CREATED → IN_PROGRESS on the first ask). No answer was graded, so
     * readiness is left as it was.
     */
    async nextQuestion (): Promise<void> {
      if (this.activeId === null || this.busy) return
      this.busy = true
      this.error = null
      try {
        const turn = await apiPost<InterviewTurn>(
          interviewSessionPath(V2_ENDPOINTS.interviewNextQuestion, this.activeId))
        this.session = turn.session
        this.question = turn.question
        this.lastQuestion = null
        this.lastAnswer = null
        this.lastEvaluation = null
        this.patchSession()
      }
      catch (caught) {
        this.error = errorMessage(caught)
      }
      finally {
        this.busy = false
      }
    },

    /** Submit a typed answer to the current question, then refresh the platform readiness. */
    async submitText (content: string): Promise<void> {
      const text = content.trim()
      if (text === '' || this.activeId === null || this.busy) return
      this.busy = true
      this.error = null
      try {
        const outcome = await apiPost<InterviewAnswerOutcome>(
          interviewSessionPath(V2_ENDPOINTS.interviewAnswers, this.activeId), { content: text })
        this.applyOutcome(outcome)
        await this.refreshReadiness()
      }
      catch (caught) {
        this.error = errorMessage(caught)
      }
      finally {
        this.busy = false
      }
    },

    // VOICE-MARKER

    /**
     * Submit a spoken answer: the clip goes up as multipart under the field name the route
     * expects (`audio`), the server transcribes it, records it as `VOICE` and grades it.
     *
     * The transcript is the server's, never the client's — the store sends bytes and reads
     * back the answer it was turned into, with its transcript confidence.
     */
    async submitVoice (clip: Blob, filename = 'answer.webm'): Promise<void> {
      if (this.activeId === null || this.busy) return
      this.busy = true
      this.error = null
      try {
        const form = new FormData()
        form.append('audio', clip, filename)
        const outcome = await apiPostForm<InterviewAnswerOutcome>(
          interviewSessionPath(V2_ENDPOINTS.interviewVoiceAnswers, this.activeId), form)
        this.applyOutcome(outcome)
        await this.refreshReadiness()
      }
      catch (caught) {
        this.error = errorMessage(caught)
      }
      finally {
        this.busy = false
      }
    },

    /**
     * Strictly re-grade the last answered question.
     *
     * A submit degrades quietly when the provider cannot coach — the answer is kept, the
     * evaluation is null. This asks again and, unlike the submit, surfaces the provider's
     * failure (503 → `evaluation_unavailable`) rather than swallowing it. The sequence is
     * the answered question's, held in `lastQuestion` for exactly this.
     */
    async evaluate (): Promise<void> {
      if (this.activeId === null || this.lastQuestion === null || this.busy) return
      this.busy = true
      this.error = null
      try {
        this.lastEvaluation = await apiPost<InterviewAnswerEvaluation>(
          interviewEvaluate(this.activeId, this.lastQuestion.sequence))
        await this.refreshReadiness()
      }
      catch (caught) {
        this.error = errorMessage(caught)
      }
      finally {
        this.busy = false
      }
    },

    // CLOSE-MARKER

    /**
     * Complete the session: finalize it and take the summary.
     *
     * The summary's readiness is the platform's computed signal, so it becomes the panel's
     * readiness verbatim. The completed session drops out of the active set and its
     * summary joins the history.
     */
    async complete (): Promise<void> {
      if (this.activeId === null || this.busy) return
      this.busy = true
      this.error = null
      try {
        const summary = await apiPost<InterviewSessionSummary>(
          interviewSessionPath(V2_ENDPOINTS.interviewComplete, this.activeId))
        this.summary = summary
        this.readiness = summary.readiness
        this.question = null
        if (this.session !== null) {
          this.session = { ...this.session, status: 'COMPLETED', is_active: false }
        }
        this.patchSession()
        await this.loadHistory()
      }
      catch (caught) {
        this.error = errorMessage(caught)
      }
      finally {
        this.busy = false
      }
    },

    /** Abandon the session — a terminal state; the practice is walked away from, not lost. */
    async abandon (): Promise<void> {
      if (this.activeId === null || this.busy) return
      this.busy = true
      this.error = null
      try {
        this.session = await apiPost<InterviewSession>(
          interviewSessionPath(V2_ENDPOINTS.interviewAbandon, this.activeId))
        this.question = null
        this.patchSession()
      }
      catch (caught) {
        this.error = errorMessage(caught)
      }
      finally {
        this.busy = false
      }
    },

    // HELPERS-MARKER

    /**
     * Record a submit's outcome: the moved session, the answer, its coaching (or null).
     *
     * The answered question becomes `lastQuestion` (its sequence is what a re-grade needs)
     * and the current question is cleared — the loop now waits on the next ask.
     */
    applyOutcome (outcome: InterviewAnswerOutcome): void {
      this.session = outcome.session
      this.lastQuestion = this.question
      this.question = null
      this.lastAnswer = outcome.answer
      this.lastEvaluation = outcome.evaluation
      this.patchSession()
    },

    /**
     * Re-fetch the open session's readiness from its own endpoint.
     *
     * Secondary to the action that triggered it: a failure here keeps the prior signal and
     * stays silent, rather than masking the grade that just succeeded with a readiness error.
     */
    async refreshReadiness (): Promise<void> {
      if (this.activeId === null) return
      try {
        this.readiness = await apiGet<SessionReadiness>(
          interviewSessionPath(V2_ENDPOINTS.interviewReadiness, this.activeId))
      }
      catch {
        // keep the last known readiness
      }
    },

    /** Reflect the open session back into the list, so its row tracks its status. */
    patchSession (): void {
      const current = this.session
      if (current === null) return
      this.sessions = this.sessions.map(session =>
        session.id === current.id ? current : session)
    },

    /** Clear everything derived from a session — used when switching the open one. */
    resetActive (): void {
      this.session = null
      this.question = null
      this.lastQuestion = null
      this.lastAnswer = null
      this.lastEvaluation = null
      this.readiness = null
      this.summary = null
      this.error = null
    },
  },
})

File: backend/app/api/routes/interview.py
"""`/api/v2/interview-sessions`: the adaptive interview simulator, over HTTP.

The phase's one rule shapes this surface: the simulator is practice and coaching, never
prediction. So the boundary the domain enforces by *absence* is enforced here too — no route
returns a hiring probability, a recruiter verdict or a provider-authored readiness. Readiness
is surfaced only on its own endpoint, computed by the platform from stored grades
(`GET /{id}/readiness`), and an answer's evaluation carries coaching, never a forecast.

The owner is never in the path or the body — it is the account resolved from the session, so
no request can open, drive, read or grade another user's practice (docs/ENGINEERING_STANDARDS.md
§Security). Reads answer 404 for "no such session" and "not yours" alike, so an id cannot be
probed. No request body carries a `user_id`, a `status`, a question's identity or an
evaluation's grade: those are the engine's and the platform's to assign.

The loop is deliberately small and idempotent: `POST /{id}/next-question` returns the current
question (polling never asks twice) or the signal to complete; the candidate answers by text
(`POST /{id}/answers`) or by voice (`POST /{id}/voice-answers`, whose audio is transcribed and
discarded — only the transcript is stored); a strict re-grade lives at
`POST /{id}/questions/{sequence}/evaluate`; and `complete` or `abandon` closes the session.
"""
from fastapi import APIRouter, File, UploadFile, status

from backend.app.api.dependencies import CurrentSession, Interviews, Now
from backend.app.api.schemas import (
    AnswerOutcomeResponse,
    CreateInterviewSessionRequest,
    InterviewAnswerEvaluationResponse,
    InterviewSessionDetailResponse,
    InterviewSessionListResponse,
    InterviewSessionResponse,
    InterviewSessionSummaryListResponse,
    InterviewSessionSummaryResponse,
    InterviewTurnResponse,
    SessionReadinessResponse,
    SubmitTextAnswerRequest,
)
from backend.app.domain.identifiers import InterviewSessionId
from backend.app.interview.transcriber import InterviewAudio

router = APIRouter(tags=["v2-interview"])


# --- opening and listing sessions -------------------------------------------

@router.post("/interview-sessions", response_model=InterviewSessionResponse,
             status_code=status.HTTP_201_CREATED)
async def create_session(body: CreateInterviewSessionRequest, current: CurrentSession,
                         service: Interviews, instant: Now) -> InterviewSessionResponse:
    """Plan and open a practice session against one posting, grounded in the account's profile.

    201, because it creates a resource. The body names the profile and posting to rehearse
    and the `mode`; the session opens `CREATED`, and the platform stamps its plan's mode so
    the session and plan can never disagree. A foreign or missing profile/posting is a 404
    (`interview_grounding_not_found`) that does not say which was missing. An `application_id`,
    when given, must be this account's and rehearse this very posting: a foreign or missing one
    is 404 (`application_not_found`), and one for a different opportunity is 409
    (`application_opportunity_mismatch`). No question is asked yet — the first `next-question`
    starts the exchange.
    """
    session = await service.create_session(
        current.user.id, candidate_profile_id=body.candidate_profile_id,
        opportunity_id=body.opportunity_id, mode=body.mode, now=instant,
        style=body.style, difficulty=body.difficulty, language=body.language,
        application_id=body.application_id, title=body.title)
    return InterviewSessionResponse.of(session)


@router.get("/interview-sessions", response_model=InterviewSessionListResponse)
async def list_sessions(current: CurrentSession,
                        service: Interviews) -> InterviewSessionListResponse:
    """This account's practice sessions, most recently updated first."""
    sessions = await service.list_sessions(current.user.id)
    return InterviewSessionListResponse.of(sessions)


@router.get("/interview-sessions/history",
            response_model=InterviewSessionSummaryListResponse)
async def readiness_history(
        current: CurrentSession,
        service: Interviews) -> InterviewSessionSummaryListResponse:
    """This account's readiness history — the summaries of completed sessions, most recent first.

    Declared before `/{session_id}` so the literal `history` is never parsed as a session id.
    Each summary pairs the platform-computed readiness with the coaching prose that explains
    it, so a candidate can watch progress over repeated practice — never a hiring forecast.
    """
    summaries = await service.readiness_history(current.user.id)
    return InterviewSessionSummaryListResponse.of(summaries)


@router.get("/interview-sessions/{session_id}",
            response_model=InterviewSessionResponse)
async def get_session(session_id: InterviewSessionId, current: CurrentSession,
                      service: Interviews) -> InterviewSessionResponse:
    """One session's configuration and lifecycle, or 404 if it is not this account's."""
    session = await service.get_session(current.user.id, session_id)
    return InterviewSessionResponse.of(session)


@router.get("/interview-sessions/{session_id}/detail",
            response_model=InterviewSessionDetailResponse)
async def session_detail(session_id: InterviewSessionId, current: CurrentSession,
                         service: Interviews) -> InterviewSessionDetailResponse:
    """One session with its whole exchange — questions, answers, evaluations and summary.

    The transcript view: everything the session holds, loaded under one ownership gate. 404
    when the session is not this account's.
    """
    detail = await service.session_detail(current.user.id, session_id)
    return InterviewSessionDetailResponse.of(detail)


@router.get("/interview-sessions/{session_id}/readiness",
            response_model=SessionReadinessResponse)
async def session_readiness(session_id: InterviewSessionId, current: CurrentSession,
                            service: Interviews, instant: Now) -> SessionReadinessResponse:
    """The session's readiness as it stands now — a coaching signal, computed by the platform.

    The same deterministic aggregation `complete` uses, run without ending the session, so a
    UI can show progress mid-practice. No provider is in this call, and the result is never a
    hiring probability: `band` is a practice label, `UNKNOWN` when nothing was evaluable rather
    than a low score. 404 when the session is not this account's.
    """
    readiness = await service.session_readiness(current.user.id, session_id, now=instant)
    return SessionReadinessResponse.of(readiness)


# --- the practice loop: ask, answer, grade ----------------------------------

@router.post("/interview-sessions/{session_id}/next-question",
             response_model=InterviewTurnResponse)
async def next_question(session_id: InterviewSessionId, current: CurrentSession,
                        service: Interviews, instant: Now) -> InterviewTurnResponse:
    """The current question to answer, or the signal the plan is done.

    Idempotent: if the last question is still unanswered it is returned again, so polling never
    asks twice; otherwise the session advances (a `CREATED` session becomes `IN_PROGRESS` on
    its first question) and the engine picks a provider-warranted follow-up or the next
    uncovered topic. When neither yields a question, the turn carries `question=null` and the
    caller completes the session. 404 when the session is not this account's, 409 when it is no
    longer active.
    """
    turn = await service.next_question(current.user.id, session_id, now=instant)
    return InterviewTurnResponse.of(turn)


@router.post("/interview-sessions/{session_id}/answers",
             response_model=AnswerOutcomeResponse,
             status_code=status.HTTP_201_CREATED)
async def submit_text_answer(session_id: InterviewSessionId,
                             body: SubmitTextAnswerRequest, current: CurrentSession,
                             service: Interviews, instant: Now) -> AnswerOutcomeResponse:
    """Record a typed answer to the current question and grade it best-effort.

    201, because it records an answer. The answer is stored and the session advances even when
    coaching could not be produced (`evaluation=null`), so a lost evaluation never costs the
    candidate their turn. 404 when the session is not this account's; 409 for an inactive
    session or an out-of-order answer.
    """
    outcome = await service.submit_text_answer(
        current.user.id, session_id, body.content, now=instant)
    return AnswerOutcomeResponse.of(outcome)


@router.post("/interview-sessions/{session_id}/voice-answers",
             response_model=AnswerOutcomeResponse,
             status_code=status.HTTP_201_CREATED)
async def submit_voice_answer(session_id: InterviewSessionId, current: CurrentSession,
                              service: Interviews, instant: Now,
                              audio: UploadFile = File(...)) -> AnswerOutcomeResponse:
    """Transcribe a spoken answer, then record and grade it exactly like a typed one.

    201, like a text answer. The session is confirmed active before a byte is transcribed, so
    a dead session spends no transcription; the raw audio is transcribed and discarded, and
    only the transcript is stored and graded. An upload larger than the transcriber accepts is
    413, an unsupported media type is 415, and a transcriber that is unavailable or returns an
    empty transcript is 503 — the audio bytes are never echoed back. A transcript whose
    confidence falls below the auto-evaluate threshold is 422 (`transcript_review_required`),
    carrying the transcript for the candidate to review and resubmit as a text answer: nothing
    is recorded and readiness is untouched, so speech-to-text uncertainty never grades an
    answer. The upload's declared content type is passed through as-is for the transcriber to
    validate.
    """
    content = await audio.read()
    payload = InterviewAudio(content=content, content_type=audio.content_type or "")
    outcome = await service.submit_voice_answer(
        current.user.id, session_id, payload, now=instant)
    return AnswerOutcomeResponse.of(outcome)


@router.post("/interview-sessions/{session_id}/questions/{sequence}/evaluate",
             response_model=InterviewAnswerEvaluationResponse)
async def evaluate_answer(session_id: InterviewSessionId, sequence: int,
                          current: CurrentSession, service: Interviews,
                          instant: Now) -> InterviewAnswerEvaluationResponse:
    """(Re)grade one answered question, surfacing failure as an error rather than swallowing it.

    The strict counterpart to the best-effort grading a submit does: where a submit keeps an
    un-gradable answer and reports `evaluation=null`, this is the path to retry coaching for an
    answer that lacked it, and it answers 503 (`evaluation_unavailable`) when the provider
    fails or its coaching does not survive the evidence guard. It does not adapt difficulty, so
    a retry never double-counts. 404 when no question holds that sequence; 409 when the question
    has no answer to grade. The evaluation carries coaching only — never a readiness or verdict.
    """
    evaluation = await service.evaluate_answer(
        current.user.id, session_id, question_sequence=sequence, now=instant)
    return InterviewAnswerEvaluationResponse.of(evaluation)


# --- closing a session ------------------------------------------------------

@router.post("/interview-sessions/{session_id}/complete",
             response_model=InterviewSessionSummaryResponse)
async def complete_session(session_id: InterviewSessionId, current: CurrentSession,
                           service: Interviews,
                           instant: Now) -> InterviewSessionSummaryResponse:
    """Aggregate readiness, write the closing summary, and finalize the session.

    Readiness is computed by the platform from stored grades — never a provider — and paired
    with coaching prose the guard clears; if the prose fails, the session still completes with
    a safe, fact-free headline, and the readiness stands either way. `IN_PROGRESS → COMPLETED`;
    a session not in progress is a 409 (`invalid_status_transition`). 404 when it is not this
    account's.
    """
    summary = await service.complete_session(current.user.id, session_id, now=instant)
    return InterviewSessionSummaryResponse.of(summary)


@router.post("/interview-sessions/{session_id}/abandon",
             response_model=InterviewSessionResponse)
async def abandon_session(session_id: InterviewSessionId, current: CurrentSession,
                          service: Interviews, instant: Now) -> InterviewSessionResponse:
    """Walk away from a session, moving it to the terminal `ABANDONED` state.

    A `CREATED` or `IN_PROGRESS` session becomes `ABANDONED`; no summary is written, because an
    abandoned session has no closing coaching. A session already terminal is a 409
    (`invalid_status_transition`); 404 when it is not this account's.
    """
    session = await service.abandon_session(current.user.id, session_id, now=instant)
    return InterviewSessionResponse.of(session)
